metadata-load/src/main/java/uk/ac/ebi/ampt2d/metadata/importer/api/SraObjectsImporterThroughAPI.java

```diff
@@ -0,0 +1,115 @@
+import uk.ac.ebi.ampt2d.metadata.persistence.repositories.StudyRepository;
+import uk.ac.ebi.ena.sra.xml.AnalysisType;
+import uk.ac.ebi.ena.sra.xml.LinkType;
+import uk.ac.ebi.ena.sra.xml.StudyType;
+import uk.ac.ebi.ena.sra.xml.XRefType;
+
+import java.util.HashSet;
+import java.util.Set;
+import java.util.stream.Stream;
+
+public class SraObjectsImporterThroughAPI extends ObjectsImporter {
+
+    public SraObjectsImporterThroughAPI(SraXmlRetrieverThroughApi sraXmlRetrieverThroughApi,
+                                        SraXmlParser<StudyType> sraStudyXmlParser,
+                                        Converter<StudyType, Study> studyConverter,
+                                        PublicationExtractorFromStudy publicationExtractorFromStudy,
+                                        WebResourceExtractorFromStudy webResourceExtractorFromStudy,
+                                        TaxonomyExtractor taxonomyExtractor,
+                                        SraXmlParser<AnalysisType> sraAnalysisXmlParser,
+                                        Converter<AnalysisType, Analysis> analysisConverter,
+                                        FileExtractorFromAnalysis fileExtractorFromAnalysis,
+                                        AnalysisRepository analysisRepository,
+                                        StudyRepository studyRepository) {
+        super(sraXmlRetrieverThroughApi, sraStudyXmlParser, sraAnalysisXmlParser, studyConverter, analysisConverter,
+                publicationExtractorFromStudy, webResourceExtractorFromStudy, taxonomyExtractor,
+                fileExtractorFromAnalysis, analysisRepository, studyRepository);
+    }
+
+    @Override
+    protected Study extractAnalysisFromStudy(StudyType studyType, Study study) {
+        studyRepository.save(study);
+        for (String analysisAccession : getAnalysisAccessions(studyType)) {
+            Analysis analysis = importAnalysis(analysisAccession);
+            analysis.setStudy(study);
+            analysisRepository.save(analysis);
+        }
+        return study;
+    }
+
+    @Override
+    protected Analysis extractStudyFromAnalysis(AnalysisType analysisType, Analysis analysis) {
+        return analysis;
+    }
+
+    private Set<String> getAnalysisAccessions(StudyType studyType) {
+        Set<String> analysisAccessions = new HashSet<>();
+
+        StudyType.STUDYLINKS studylinks = studyType.getSTUDYLINKS();
+        if (studylinks == null) {
+            return analysisAccessions;
+        }
+        LinkType[] studyLinkArrays = studylinks.getSTUDYLINKArray();
+        String analyses = null;
+        for (int i = 0; i < studyLinkArrays.length; i++) {
+            XRefType xRefType = studyLinkArrays[i].getXREFLINK();
+            if (xRefType != null && xRefType.getDB().equals("ENA-ANALYSIS")) {
+                analyses = xRefType.getID();
+                break;
+            }
+        }
+        if (analyses == null) {
```

Comment:
Rather than having 3 returns, you can make if (analyses != null ) {stream ...} at the end return analysisAccessions;
First 2 returns have blank, but combining them will take unnecessary execution

Reply:
thanks, but I guess the code right now looks meaningful to return as soon as possible.